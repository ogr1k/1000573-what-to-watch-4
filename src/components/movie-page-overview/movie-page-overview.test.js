import React from "react";
import renderer from "react-test-renderer";
import MoviePageOverview from "./movie-page-overview.jsx";
import {Router} from "react-router-dom";
import history from "../../history.js";

const film = {
  description: `A former Prohibition-era Jewish gangster returns to the Lower East Side of Manhattan over thirty years later, where he once again must confront the ghosts and regrets of his old life.`,
  director: `Sergio Leone`,
  genre: `Crime`,
  id: 1,
  name: `Once Upon a Time in America`,
  poster: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Once_Upon_a_Time_in_America.jpg`,
  previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/Once_Upon_a_Time_in_America.jpg`,
  rating: 9.9,
  year: 1984,
  ratings: 276395,
  starring: [`Robert De Niro`, `James Woods`, `Elizabeth McGovern`],
  video: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  runTime: 500
};


it(`Render Movie Page Overview`, () => {

  const tree = renderer
    .create(

        <Router history={history}>
          <MoviePageOverview
            activeTab={``}
            film={film}
          />
        </Router>

    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
