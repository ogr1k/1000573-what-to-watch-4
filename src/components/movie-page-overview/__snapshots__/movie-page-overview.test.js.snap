// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Movie Page Overview 1`] = `
<div
  className="movie-card__wrap movie-card__translate-top"
>
  <div
    className="movie-card__info"
  >
    <div
      className="movie-card__poster movie-card__poster--big"
    >
      <img
        alt="The Grand Budapest Hotel poster"
        height={327}
        src="https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Once_Upon_a_Time_in_America.jpg"
        width={218}
      />
    </div>
    <div
      className="movie-card__desc"
    >
      <nav
        className="movie-nav movie-card__nav"
      >
        <ul
          className="movie-nav__list"
        >
          <li
            className="movie-nav__item"
          >
            <a
              className="movie-nav__link"
              href="/films/1"
              onClick={[Function]}
            >
              Overview
            </a>
          </li>
          <li
            className="movie-nav__item"
          >
            <a
              className="movie-nav__link"
              href="/films/1/details"
              onClick={[Function]}
            >
              Details
            </a>
          </li>
          <li
            className="movie-nav__item"
          >
            <a
              className="movie-nav__link"
              href="/films/1/review"
              onClick={[Function]}
            >
              Reviews
            </a>
          </li>
        </ul>
      </nav>
      <div
        className="movie-rating"
      >
        <div
          className="movie-rating__score"
        >
          9.9
        </div>
        <p
          className="movie-rating__meta"
        >
          <span
            className="movie-rating__level"
          >
            Very good
          </span>
          <span
            className="movie-rating__count"
          >
            276395
             ratings
          </span>
        </p>
      </div>
      <div
        className="movie-card__text"
      >
        <p>
          A former Prohibition-era Jewish gangster returns to the Lower East Side of Manhattan over thirty years later, where he once again must confront the ghosts and regrets of his old life.
        </p>
        <p
          className="movie-card__director"
        >
          <strong>
            Director: 
            Sergio Leone
          </strong>
        </p>
        <p
          className="movie-card__starring"
        >
          <strong>
            Starring: 
            Robert De Niro, James Woods, Elizabeth McGovern
             and other
          </strong>
        </p>
      </div>
    </div>
  </div>
</div>
`;
